### DISCUSSION TITLE
Sprint 4 Demo Discussion

### DISCUSSION THREAD

**Discussion by Coffee2Bits** [DevOps-MBSE/AaC#133] [Archive]
Here are the issues that were successfully completed during Sprint 4:

* #113 
* #7 
* #38 
* #93
* #99 
* #109 

Topics/Flow:
### Plugin with Extensions only
Updated `gen-plugin` command to allow generating a plugin with no `behavior`s - which means it has no commands. This allows users to create a plugin that extends the AaC spec without those extensions having to be part of the spec. 

### Moved the AaC Spec into the Package
Embedded source-of-truth documents (yaml) and other data files (jinja2 templates) into the project's distribution. We have also created some functional tests for our package distribution before it's deployed to the Python Package Index. These functional tests take the `wheel` distribution, the file that will be uploaded to PyPI, and they install the AaC Package's `wheel` in a fresh environment and they execute the package's CLI commands, just like a user would call.

### Pipeline Updates
* Added a new job `Package-Functional-Tests`
* Testing our wheel distribution package like a user would, by installing the wheel and running aac commands 

The pipeline received some much needed TLC. In order to support more complex testing of the AaC's distribution, the pipeline was broken out into two jobs. The first job focuses on build, unit testing, linting, generating documentation, and running dependency analysis. The second job focuses on testing the AaC package before it's deployed to PyPI, by running some functional tests on the package installed from the distribution wheel in a clean test environment.

Some other work on the pipeline include:
1. Dependency tree scanning
2. Addressing technical debt in the testing environment

### UI MVP

**Comment by lizzcondrey** [comment]
Closing since this is marked as resolved and the sprint is past.
